docs/source/federated-types/federated-directives.mdx

```diff
@@ -0,0 +1,302 @@
+---
+title: Federation-specific GraphQL directives
+---
+
+Apollo Federation defines a collection of **directives** that you use in your subgraph schemas to enable certain features.
+
+## Importing directives
+
+To use federated directives in a Federation 2 subgraph schema, apply the `@link` directive with the following format to the `schema` type:
+
+```graphql
+extend schema
+  @link(url: "https://specs.apollo.dev/federation/v2.0",
+        import: ["@key", "@shareable"])
+```
+
+You can apply this directive to your existing `schema` declaration if you have one, or to a new `extend schema` declaration (as shown above).
+
+Modify the `import` array to include whichever federated directives your subgraph schema uses. The example above imports the [`@key`](#key) and [`@shareable`](#shareable) directives (which are used most commonly).
+
+> Make sure to include the `@` in each directive name!
+
+### Renaming directives
+
+If a federated directive's name matches one of your own custom directives, you can rename the federated directive with the following syntax:
+
+```graphql
+extend schema
+  @link(url: "https://specs.apollo.dev/federation/v2.0",
+        import: [{ name: "@key", as: "@primaryKey"}, "@shareable"])
+```
+
+This example subgraph schema uses `@primaryKey` for the federated directive usually named [`@key`](#key).
+
+## Managing entities
+
+### `@key`
+
+Designates an object type as an [entity](../entities/).
+
+**Valid locations:** Object type definitions
+
+```graphql {1}
+type Product @key(fields: "id") {
+  id: ID!
+  name: String!
+  price: Int
+}
+```
+
+#### Arguments
+
+<table class="field-table">
+  <thead>
+    <tr>
+      <th>Name /<br/>Type</th>
+      <th>Description</th>
+    </tr>
+  </thead>
+
+<tbody>
+
+<tr class="required">
+<td>
+
+##### `fields`
+
+`String`
+</td>
+<td>
+
+**Required.** A selection set of field names (and optionally _subfield_ names) that contribute to the entity's primary key.
+
+Examples:
+
+* `"id"`
+* `"username region"`
+* `"name organization { id }"`
+
+See also [Advanced `@key`s](../entities-advanced/#advanced-keys).
+
+</td>
+</tr>
+
+<tr>
+<td>
+
+##### `resolvable`
+
+`Boolean`
+</td>
+<td>
+
+If `false`, informs the gateway that this subgraph _doesn't_ define a [reference resolver](../entities/#2-define-a-reference-resolver) for this entity. This means that gateway query plans can't "jump to" this subgraph to resolve fields that aren't defined in _another_ subgraph.
+
+Most commonly, you set this to `false` when [referencing an entity without contributing fields](../entities/#referencing-an-entity-without-contributing-fields).
+
+The default value is `true`.
+
+</td>
+</tr>
+
+</tbody>
+</table>
+
+### `@override`
+
+Indicates that a non-`@shareable` entity field is now resolved by _this_ subgraph instead of the _one other_ subgraph where it's currently defined.
```

Comment:
You can remove `non-@shareable`: it's possible to override a shareable field if one wants to. In that case, the specifically pointed overridden field will never be resolved anymore but both the newly added one (the one with the `@override` and any other unrelated definitions of that field may be resolved, at the discretion of the query planner (as is true for normal shareable entity fields).

Essentially, `@override` just says "pretend that other definition I'm pointing to doesn't exists", but there is no specific other constraint. In fact, you can use `@override` on a non-entity field if you want, and that could actually be useful if you want to move a top-level operation (a `Query` field basically) between subgraphs.

Reply:
Cool thanks! Let's say I have a `@shareable` field and I want to `@override` two _different_ subgraphs. Can I apply two `@override` directives to the same field? Or do I somehow include both subgraph names in one `from` argument? Or is this not possible currently?

Reply:
Currently no, it's not possible: `@override` is not repeatable and `from` cannot have multiple options. I considered it, but decided to leave that out initially. I couldn't come up with a realistic scenario where it would be that useful, and I didn't wanted that to introduce new unforeseen problem just because we haven't taken the time to think it through.

But I "think" it's pretty trivial to add once we want to by allowing `@override` to be repeatable.

There is a few other limits we currently impose to make it simply to reason about, like you cannot "chain" overrides (overriding a field that itself override another subgraph) currently, but in theory I think we could lift that limitation if good use cases for it comes.